### PR TITLE
Remove the OCap system

### PR DESCRIPTION
**Pull request by plafer**
Closes: #2159
Closes: #2116

## Description
This PR removes everything "object capability" in the modules. Notably, there is now only 1 function to bind a module to a port (`PortKeeper::bind_module_to_port(module_id, port_Id)`), and 1 function to lookup a module by port (`PortReader::lookup_module_by_port(port_id)`).

We currently don't support [transferring ownership of a port](https://github.com/cosmos/ibc/blob/master/spec/core/ics-005-port-allocation/README.md#transferring-ownership-of-a-port) or [releasing a port](https://github.com/cosmos/ibc/blob/master/spec/core/ics-005-port-allocation/README.md#releasing-a-port).

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [x] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: modules/README.md [hunk 1]
```diff
@@ -18,6 +18,31 @@ Implementation of the Inter-Blockchain Communication Protocol ([IBC]) module.
 
 See documentation on [docs.rs][docs-link].
 
+## Divergence from the Interchain Standards (ICS)
```
Comment (adizere) [inline modules/README.md:21]: Tip-top, thanks for taking the care to describe this!

**Review by adizere — APPROVED** [review]
This is a very satisfying PR! Pre-approved.

File: modules/src/core/ics04_channel/handler/acknowledgement.rs [hunk 2]
```diff
@@ -195,23 +193,12 @@ mod tests {
...
-                msg: msg.clone(),
-                want_pass: false,
-            },
             Test {
                 name: "Good parameters".to_string(),
                 ctx: context
                     .with_client(&ClientId::default(), client_height)
                     .with_connection(ConnectionId::default(), connection_end)
-                    .with_port_capability(packet.destination_port.clone())
+                    .with_port(packet.destination_port.clone())
```
Comment (hu55a1n1) [inline modules/src/core/ics04_channel/handler/acknowledgement.rs:201]: IIUC, I don't think we need to set a port mapping to test handshakes, so we should be able to remove most of them.

**Review by hu55a1n1 — COMMENTED** [review]
Thanks, @plafer! 🙏 There's nothing like a PR that has more deletions than additions! I added some suggestions and comments.

File: modules/src/mock/context.rs [hunk 3]
```diff
@@ -527,13 +524,11 @@ impl MockContext {
 
     pub fn add_port(&mut self, port_id: PortId) {
         let module_id = ModuleId::new(format!("module{}", port_id).into()).unwrap();
```
[the remark this answers is not in the document]
Comment (plafer) [reply modules/src/mock/context.rs:526]: Also referenced in [issue](https://github.com/informalsystems/ibc-rs/issues/2209)

**Pull request merged**